src/Npgsql/NpgsqlDataSourceBuilder.cs

```diff
@@ -153,6 +155,28 @@ public NpgsqlDataSourceBuilder UseClientCertificatesCallback(Action<X509Certific
         return this;
     }
 
+    /// <summary>
+    /// Sets the <see cref="X509Certificate2" /> that will be used validate SSL certificate, received from the server.
+    /// </summary>
+    /// <param name="validationRootCertificate">The CA certificate.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlDataSourceBuilder UseValidationRootCertificate(X509Certificate2? validationRootCertificate) 
+        => validationRootCertificate is null 
+            ? UseValidationRootCertificateCallback(null) 
+            : UseValidationRootCertificateCallback(() => validationRootCertificate);
+
+    /// <summary>
+    /// Specifies a callback that will be used to validate SSL certificate, received from the server.
+    /// </summary>
+    /// <param name="validationRootCertificateCallback">The callback to get CA certificate.</param>
+    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
+    public NpgsqlDataSourceBuilder UseValidationRootCertificateCallback(Func<X509Certificate2>? validationRootCertificateCallback)
```

Comment:
I think the reason we accept a callback for the cilent certificate (in addition to the overload that just receives it) is to support rotating it; do we think that's really relevant for the root CA certificate?

In any case, we should probably add a note in the XML docs saying that these overloads are for rotation scenarios only. Something like: "This overload, which accepts a callback, is suitable for scenarios where the certificate rotates and might change during the lifetime of the application. When that's not the case, use the overload which directly accepts the certificate."

Reply:
> I think the reason we accept a callback for the cilent certificate (in addition to the overload that just receives it) is to support rotating it; do we think that's really relevant for the root CA certificate?

Maybe? Mostly it's just to have the exact same API as client certificate has which isn't that bad of a reason on itself.

Reply:
Added a note, the one for client certificate already has something similar.